doc/source/development/contributing_code_guide.rst

```diff
@@ -0,0 +1,140 @@
+.. _contributing_code_guide:
+
+{{ header }}
+
+=======================
+pandas code style guide
+=======================
+
+.. contents:: Table of contents:
+   :local:
+
+Patterns
+========
+
+foo.__class__
+-------------
+
+*pandas* uses 'type(foo)' instead 'foo.__class__' as it is making the code more
+readable.
+
+For example:
+
+**Good:**
+
+.. code-block:: python
+
+    class MyClass:
+        def __init__(self):
+            cls = type(self)
```

Comment:
I think all these examples are too complicated to illustrate the point here.

You can simply show something like:
```python
foo = 'bar'
type(foo)  # good
foo.__class__  # bad
```

I think creating classes, too many cases... creates noise, more than help understand the point.